### PR TITLE
CNF-13756: Adjust mustgather test suite for hypershift

### PR DESCRIPTION
**Pull request by SargunNarula**
It includes adjustments required to run the mustgather suite on a hypershift hosted cluster.

### CODE REVIEW

File: test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go [hunk 1]
```diff
@@ -149,7 +156,7 @@ var _ = Describe("[rfe_id: 50649] Performance Addon Operator Must Gather", Label
 			Expect(err).ToNot(HaveOccurred())
 		})
 
-		It("Verify machineconfig resources are captured", func() {
+		It("Verify machineconfig resources are captured", Label(string(label.OpenShift)), func() {
 			mcps := &machineconfigv1.MachineConfigPoolList{}
 			err := testclient.Client.List(context.TODO(), mcps)
```
Comment (Tal-or) [inline test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go:161]: Replace `Client` with `ControlPlaneClient`.
`Client` is obsolete.

**Comment by Tal-or** [comment]
Those tests are not actually running we should add them to the e2e make target

File: test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go [hunk 2]
```diff
@@ -62,10 +63,16 @@ var _ = Describe("[rfe_id: 50649] Performance Addon Operator Must Gather", Label
...
-				"cluster-scoped-resources/machineconfiguration.openshift.io/kubeletconfigs/performance-performance.yaml",
 				"namespaces/openshift-cluster-node-tuning-operator/tuned.openshift.io/tuneds/openshift-node-performance-performance.yaml",
 			}
+			if !hypershiftutils.IsHypershiftCluster() {
+				ClusterSpecificFiles = append(ClusterSpecificFiles,
+					"cluster-scoped-resources/performance.openshift.io/performanceprofiles/performance.yaml",
+					"cluster-scoped-resources/machineconfiguration.openshift.io/kubeletconfigs/performance-performance.yaml",
+				)
+			} else {
+				ClusterSpecificFiles[0] = "namespaces/openshift-cluster-node-tuning-operator/tuned.openshift.io/tuneds/openshift-node-performance-performance-*.yaml"
```
[the remark this answers is not in the document]
Comment (SargunNarula) [reply test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go]: Improved, Thanks

File: test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go [hunk 3]
```diff
@@ -166,8 +172,17 @@ var _ = Describe("[rfe_id: 50649] Performance Addon Operator Must Gather", Label
...
+		pattern, err := filepath.Glob(file)
+		if err != nil {
+			// if the pattern itself is malformed.
+			return fmt.Errorf("failed to match pattern for file: %s, error: %w", file, err)
+		}
+		if pattern == nil {
+			// When no files matched the pattern.
+			return fmt.Errorf("no files matched the pattern for file: %s", file)
+		}
+		if _, err = os.Stat(pattern[0]); errors.Is(err, os.ErrNotExist) {
```
Comment (Tal-or) [inline test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go:184]: you are not covering the case when error is not nil but also not  os.ErrNotExist

File: test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go [hunk 4]
```diff
@@ -166,8 +172,17 @@ var _ = Describe("[rfe_id: 50649] Performance Addon Operator Must Gather", Label
...
+			return fmt.Errorf("file does not exist: %s", pattern[0])
```
Comment (Tal-or) [inline test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go:185]: please return the original error as well

**Comment by Tal-or** [comment]
/test e2e-hypershift 
infra issue

File: test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go (hunk 3 above)
Comment (SargunNarula) [reply test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go]: Resolved.

**Comment by Tal-or** [comment]
/lgtm
Thank you for working on this

**Pull request merged**